app/Decsys/Services/WebhookService.cs

```diff
@@ -21,5 +26,69 @@ IWebhookRepository webhooks
     /// <returns>The created Webhook</returns>
     public int Create(WebhookModel model)
         => _webhooks.Create(model);
+
     
+    /// <summary>
+    /// Triggers webhooks from a given payload.
+    /// </summary>
+    /// <param name="payload">The payload to trigger and Post.</param>
+    public async Task Trigger(PayloadModel payload)
+    {
+        var webhooks = _webhooks.GetWebhooksBySurvey(payload.SurveyId);
+        
+        foreach (var webhook in webhooks)
+        {
+            if (await FilterCriteria(payload, webhook))
+            {
+                await SendWebhook(payload, webhook.CallbackUrl);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Filters the webhooks based on matching criteria.
+    /// Currently focussed on matching page numbers on page navigation.
+    /// </summary>
+    /// <param name="payload">Payload to check</param>
+    /// <param name="webhook">Webhook to match against.</param>
+    /// <returns>True if filter matches</returns>
+    private static Task<bool> FilterCriteria(PayloadModel payload, WebhookModel webhook)
+    {
+        foreach (var eventType in webhook.TriggerCriteria.SelectMany(filter => filter.EventTypes))
+        {
+            switch (payload.EventType)
```

Comment:
I think the switch statement here is a little unclear; a refactored version as discussed: https://dotnetfiddle.net/RtrC9A

Reply:
Have implemented this pattern